airflow/contrib/operators/gcp_container_operator.py

```diff
@@ -170,3 +175,80 @@ def execute(self, context):
         return create_op
+
+
+KUBE_CONFIG_ENV_VAR = "KUBECONFIG"
+
+
+class GKEPodOperator(KubernetesPodOperator):
+    template_fields = ('project_id', 'location',
+                       'cluster_name') + KubernetesPodOperator.template_fields
+
+    @apply_defaults
+    def __init__(self,
+                 project_id,
+                 location,
+                 cluster_name,
+                 *args,
+                 **kwargs):
+        """
+        Executes a task in a Kubernetes pod in the specified Google Kubernetes Engine
+        cluster
+
+        This Operator assumes that the system has gcloud installed and has default
+        application credentials set in order to use gcloud with the corresponding project,
+        cluster, etc.
+
+        The **minimum** required to define a cluster to create are the variables
+        ``task_id``, ``project_id``, ``location``, ``cluster_name``, ``name``,
+        ``namespace``, and ``image``
+
+        **Operator Creation**: ::
+
+            operator = GKEPodOperator(task_id='pod_op',
+                                      project_id='my-project',
+                                      location='us-central1-a',
+                                      cluster_name='my-cluster-name',
+                                      name='task-name',
+                                      namespace='default',
+                                      image='perl')
+
+        .. seealso::
+            For more detail about application authentication have a look at the reference:
+            https://cloud.google.com/docs/authentication/production#providing_credentials_to_your_application
+
+        This does not require a gcp_conn_id nor api_version since it does not end up
+        getting a connection from a hook, nor using a specific api version
+
+        :param project_id: The Google Developers Console project id
+        :type project_id: str
+        :param location: The name of the Google Kubernetes Engine zone in which the
+            cluster resides, e.g. 'us-central1-a'
+        :type location: str
+        :param cluster_name: The name of the Google Kubernetes Engine cluster the pod
+            should be spawned in
+        :type cluster_name: str
+        """
+        super(GKEPodOperator, self).__init__(*args, **kwargs)
+        self.project_id = project_id
+        self.location = location
+        self.cluster_name = cluster_name
+
+    def execute(self, context):
+        # Write config to a temp file and set the environment variable to point to it.
+        # This is to avoid race conditions of reading/writing a single file
+        with tempfile.NamedTemporaryFile() as conf_file:
+            os.environ[KUBE_CONFIG_ENV_VAR] = conf_file.name
+            # Attempt to get/update credentials
+            # We call gcloud directly instead of using google.cloud python api
+            # because google.cloud depends on gcloud to be installed anyway.
```

Comment:
"google.cloud depends on gcloud to be installed anyway" This is false.

Reply:
You can definitely use the `google-auth` and `google-cloud-*` packages without having the Google Cloud SDK (gcloud command) installed.

For consistency with other operators and since I don't think there's a `google-cloud-kubernetes-engine` package, I'd use the `google-api-python-client` for this.